src/test/java/org/javarosa/core/model/instance/geojson/GeoJsonExternalInstanceTest.java

```diff
@@ -0,0 +1,151 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.javarosa.core.model.instance.geojson;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.javarosa.core.model.instance.TreeElement;
+import org.junit.Test;
+
+public class GeoJsonExternalInstanceTest {
+    @Test
+    public void parse_throwsException_ifTopLevelElementNotFeatureCollection() {
+        try {
+            GeoJsonExternalInstance.parse("id", new ByteArrayInputStream(GEOMETRY_COLLECTION_GEOJSON.getBytes(StandardCharsets.UTF_8)));
+            fail("Exception expected");
+        } catch (IOException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void parse_parsesMultipleFeatures() throws IOException {
+        TreeElement featureCollection = GeoJsonExternalInstance.parse("id", new ByteArrayInputStream(FEATURE_COLLECTION_GEOJSON.getBytes(StandardCharsets.UTF_8)));
+        assertThat(featureCollection.getNumChildren(), is(2));
+    }
+
+    @Test
+    public void parse_addsGeometryAsChild() throws IOException {
+        TreeElement featureCollection = GeoJsonExternalInstance.parse("id", new ByteArrayInputStream(FEATURE_COLLECTION_GEOJSON.getBytes(StandardCharsets.UTF_8)));
+        assertThat(featureCollection.getChildAt(0).getChild("geometry", 0).getValue().getValue(), is("102 0.5 0 0"));
+        assertThat(featureCollection.getChildAt(1).getChild("geometry", 0).getValue().getValue(), is("104 0.5 0 0"));
+    }
+
+    @Test
+    public void parse_addsAllOtherPropertiesAsChildren() throws IOException {
+        TreeElement featureCollection = GeoJsonExternalInstance.parse("id", new ByteArrayInputStream(FEATURE_COLLECTION_GEOJSON.getBytes(StandardCharsets.UTF_8)));
+        assertThat(featureCollection.getChildAt(0).getNumChildren(), is(4));
+        assertThat(featureCollection.getChildAt(0).getChild("name", 0).getValue().getValue(), is("My cool point"));
+
+        assertThat(featureCollection.getChildAt(1).getNumChildren(), is(5));
+        assertThat(featureCollection.getChildAt(1).getChild("special-property", 0).getValue().getValue(), is("special value"));
+    }
+
+    @Test
+    public void getOdkCoordinates_convertsPointGeoJsonGeometry() throws IOException {
+        String point = "{ \"type\": \"Point\", \"coordinates\": [ 102, 0.5 ] }";
+        ObjectMapper objectMapper = new ObjectMapper();
+        GeojsonGeometry geometry = objectMapper.readValue(point, GeojsonGeometry.class);
+        assertThat(geometry.getOdkCoordinates(), is("102 0.5 0 0"));
+    }
+
+    @Test
+    public void getOdkCoordinates_throwsException_ifGeometryNotPoint() throws IOException {
```

Comment:
This test (and the one above) should live in `GeojsonGeometryTest` rather than here I reckon. Alternatively, you don't need `getOdkCoordinates_convertsPointGeoJsonGeometry` (as it's covered by your `parse` tests) and `getOdkCoordinates_throwsException_ifGeometryNotPoint` could be covered here with a test that provides JSON that will cause the fail. The latter is more correct™️ in my opinion, as it would ensure that the exception bubbles up correctly.